modal/sandbox.py

```diff
@@ -400,12 +400,12 @@ async def _get_task_id(self):
     async def exec(
         self,
         *cmds: str,
-        # Deprecated: internal use only
-        pty_info: Optional[api_pb2.PTYInfo] = None,
+        pty_info: Optional[api_pb2.PTYInfo] = None,  # Deprecated: internal use only
         stdout: StreamType = StreamType.PIPE,
         stderr: StreamType = StreamType.PIPE,
-        # Internal option to set terminal size and metadata
-        _pty_info: Optional[api_pb2.PTYInfo] = None,
+        text: bool = True,  # Encode output as text
+        by_line: bool = False,  # Line-buffered output
```

Comment:
See above comment about potentially skipping the `text` argument here. Users can achieve the same behavior by just `.encode("utf8")`:ing the output instead, with the benefit of better type safety.

I'd also be in favor of keeping this a closely mirrored to the `subprocess.Popen` interface as possible, so perhaps `by_line` could be replaced by `bufsize` with only allowed values being `-1` ("auto") or `1` (line buffered), at least in the public-facing interface

Reply:
Also, should we add the same argument(s) to Sandbox.create too to keep them mirrored?